npm/private/utils.bzl

```diff
@@ -66,26 +66,10 @@ def _parse_pnpm_package_key(pnpm_name, pnpm_version):
     if pnpm_version.startswith("link:") or pnpm_version.startswith("file:"):
         return pnpm_name, "0.0.0"
 
-    if not pnpm_version.startswith("/"):
-        if not pnpm_name:
-            fail("parse_pnpm_package_key: pnpm_name is empty for non-versioned package %s" % pnpm_version)
+    if pnpm_version.startswith("npm:"):
+        return pnpm_version[4:].rsplit("@", 1)
 
-        return pnpm_name, pnpm_version
-
-    # Parse a package key such as:
```

Comment:
These are all handled and normalized at parse-time, most from before this PR and now the final `npm:` case.